js/NavigationTitleView.js

```diff
@@ -0,0 +1,50 @@
+import Adapt from 'core/js/adapt';
+import device from 'core/js/device';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { templates } from 'core/js/reactHelpers';
+
+class NavigationTitleView extends Backbone.View {
+  className() {
+    return 'navigation-title';
+  }
+
+  attributes() {
+    return {
+      'data-order': (Adapt.course.get('_globals')?._extensions?._navigationTitle?._navOrder || 0)
+    };
+  }
+
+  initialize() {
+    this.listenTo(Adapt, 'device:changed', this.changed);
```

Comment:
Given that you want to amend model properties it might be good for clarity to have something like:

```js
this.listenTo(Adapt, 'device:changed', this.onDeviceChanged);
```

```js
onDeviceChanged() {
    this.setIsDeviceSmall();
}
```

which simplifies `changed` to only be concerned with rendering:

```js
changed() {
    ReactDOM.render(<templates.navigationTitle {...this.model.toJSON()} />, this.el);
}
```

In `initialize` add a listener so that you can render on model changes:

```js
this.listenTo(this.model, 'all', this.changed);
```

Finally, remove `hideForMobile` and make its changes directly in the React component.

Reply:
Thanks, @chris-steele.

So, I think this is fine as-is for the following reasons.

1. I purposely moved `_hideForMobile` to `NavigationTitleView.js` because I wanted to hide the parent DOM element `.navigation-logo` rather than just the child, inner div.  Leaving this element visible can create issues with Flexbox layouts and other styles as it still takes up space.  This issue was brought up in https://github.com/cgkineo/adapt-navigationLogo/issues/6 

2. `_hideForMobile` will not change.  So, we really are just rerendering when the _device size_ changes rather than any specific model properties.

Thoughts?

Reply:
Ah I see, yes, the Backbone/React hybrid does present this rather annoying problem at times with the container element. It's a simple enough plugin so if you want to leave it as is with the model manipulation in `changed` I see no harm. 👍